modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -518,23 +533,30 @@ private CompletableFuture<AsyncSqlCursor<InternalSqlRow>> querySingle(
                 : CompletableFuture.supplyAsync(() -> parseAndCache(sql), taskExecutor);
 
         return start.thenCompose(result -> {
+            queryInfo.queryType(result.queryType());
             validateParsedStatement(properties, result);
             validateDynamicParameters(result.dynamicParamsCount(), params, true);
 
             HybridTimestamp operationTime = deriveOperationTime(txContext);
 
             SqlOperationContext operationContext = SqlOperationContext.builder()
-                    .queryId(UUID.randomUUID())
+                    .queryId(queryInfo.queryId())
```

Comment:
can we obtain the value under the hood instead of explicit set?

Reply:
the simple answer is we can't, because it is only used in preparation for execution and is not present on remote nodes.